RACNest/ViewControllers/Form/FormViewController.swift

```diff
@@ -25,8 +25,8 @@ final class FormViewController: UIViewController {
         
         viewModel.username <~ usernameField.rex_textSignal
         viewModel.password <~ passwordField.rex_textSignal
-        
-        loginButton.rex_pressed <~ SignalProducer(value: viewModel.authenticate)
+
+        loginButton.rex_pressed <~ SignalProducer(value: CocoaAction(viewModel.authenticateAction) { _ in })
         
```

Comment:
You are making the VC to be aware of the `CocoaAction`, which defeats the purpose of the VM.